src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.Integer.cs

```diff
@@ -3,163 +3,254 @@
+        ///   the encoded integer is out of range for <see cref="long"/>
+        /// </exception>
+        /// <exception cref="FormatException">
+        ///   invalid CBOR encoding data --OR--
+        ///   unexpected end of CBOR encoding data --OR--
+        ///   CBOR encoding not accepted under the current conformance level
+        /// </exception>
+        public long ReadInt64()
         {
-            int value = checked((int)PeekSignedInteger(out int additionalBytes));
-            AdvanceBuffer(1 + additionalBytes);
+            long value = PeekSignedInteger(out int bytesRead);
+            AdvanceBuffer(bytesRead);
             AdvanceDataItemCounters();
             return value;
         }
 
-        public uint ReadUInt32()
+        /// <summary>
+        ///   Reads the next data item as an usigned integer (major type 0)
+        /// </summary>
+        /// <returns>The decoded integer value.</returns>
+        /// <exception cref="InvalidOperationException">
+        ///   the next data item does not have the correct major type.
+        /// </exception>
+        /// <exception cref="OverflowException">
+        ///   the encoded integer is out of range for <see cref="ulong"/>
+        /// </exception>
+        /// <exception cref="FormatException">
+        ///   invalid CBOR encoding data --OR--
+        ///   unexpected end of CBOR encoding data --OR--
+        ///   CBOR encoding not accepted under the current conformance level
+        /// </exception>
+        public ulong ReadUInt64()
         {
-            uint value = checked((uint)PeekUnsignedInteger(out int additionalBytes));
-            AdvanceBuffer(1 + additionalBytes);
+            ulong value = PeekUnsignedInteger(out int bytesRead);
+            AdvanceBuffer(bytesRead);
             AdvanceDataItemCounters();
             return value;
         }
 
-        // Returns the next CBOR negative integer encoding according to
-        // https://tools.ietf.org/html/rfc7049#section-2.1
+        /// <summary>
+        ///   Reads the next data item as a CBOR negative integer encoding (major type 1).
+        /// </summary>
+        /// <returns>
+        ///   An unsigned integer denoting -1 minus the integer.
+        /// </returns>
+        /// <exception cref="InvalidOperationException">
+        ///   the next data item does not have the correct major type.
+        /// </exception>
+        /// <exception cref="OverflowException">
+        ///   the encoded integer is out of range for <see cref="uint"/>
+        /// </exception>
+        /// <exception cref="FormatException">
+        ///   invalid CBOR encoding data --OR--
+        ///   unexpected end of CBOR encoding data --OR--
+        ///   CBOR encoding not accepted under the current conformance level
+        /// </exception>
+        /// <remarks>
+        ///   Intended as an escape hatch in cases of valid CBOR negative integers exceeding available primitive sizes.
```

Comment:
Unlike the other sections, Remarks doesn't really carry an implicit subject, so this is probably better as

```suggestion
        ///   This method provides support for reading negative numbers in the range [-9223372036854775809, -18446744073709551614],
        ///   which fit in a four-byte encoding but are outside the range of <see cref="long"/>.
        ///
        ///   The return value represents `-semanticValue + 1`, so a return value of `0` represents `-1`,
        ///   a return value of 9223372036854775810 represents -9223372036854775809,
        ///   and a return value of `18446744073709551615` represents `-18446744073709551614`.
```

Or something.  I could also be mis-remembering which way the off-by-one works :)

Reply:
Probably "values less than `long.MinValue`" would suffice?

Reply:
Well, it's not all values less than `long.MinValue`, only one-bit-more; since beyond that the BigNum encoding is required.  Since it doesn't really map to an existing type it's probably worth being explicit.  e.g. UInt32 spells out its range: https://docs.microsoft.com/en-us/dotnet/api/system.uint32?view=netcore-3.1#remarks